editor/components/inserter/menu.js

```diff
@@ -119,140 +84,127 @@ class InserterMenu extends wp.element.Component {
 		)( blockTypes );
 	}
 
-	findNext( currentBlock, blockTypes ) {
-		/**
-		 * null is the value that will trigger iterating back to
-		 * the top of the list of block types.
-		 */
-		if ( null === currentBlock ) {
-			return blockTypes[ 0 ].slug;
-		}
+	findByIncrement( blockTypes, increment = 1 ) {
+		// Add on a fake search block to the list to cycle through.
+		const list = blockTypes.concat( { slug: 'search' } );
 
-		const currentIndex = blockTypes.findIndex( ( blockType ) => currentBlock === blockType.slug );
-		const nextIndex = currentIndex + 1;
-		const highestIndex = blockTypes.length - 1;
+		const currentIndex = findIndex( list, ( blockType ) => this.state.currentFocus === blockType.slug );
```

Comment:
Good to know.  That is somewhat hard to read, especially if you didn't know lodash could do that.  Did not know about lodash.memoize either.  Thank you for the knowledge dropping.